src/viking.js

```diff
@@ -1,11 +1,86 @@
 // Soldier
-function Soldier() {}
+function Soldier(healthArg, strengthArg) {
+  this.health = healthArg;
+  this.strength = strengthArg;
+  this.attack = function () {
+    return this.strength;
+  };
+}
+
+Soldier.prototype.receiveDamage = function (damage) {
+  this.health = this.health - damage;
+};
 
 // Viking
-function Viking() {}
+function Viking(name, health, strength) {
+  Soldier.call(this, health, strength);
+  this.name = name;
+}
+
+Viking.prototype = Object.create(Soldier.prototype);
+Viking.prototype.constructor = Viking;
+Viking.prototype.receiveDamage = function (damage) {
+  this.health = this.health - damage;
+  if (this.health > 0) {
+    return this.name + " has received " + damage + " points of damage";
+  } else {
+    return this.name + " has died in act of combat";
+  }
+}
+
+Viking.prototype.battleCry = function () {
+  return "Odin Owns You All!";
+}
 
 // Saxon
-function Saxon() {}
+function Saxon(health, strength) {
+  Soldier.call(this, health, strength);
+}
+
+Saxon.prototype = Object.create(Soldier.prototype);
+Saxon.prototype.constructor = Saxon;
+Saxon.prototype.receiveDamage = function (damage) {
+  this.health = this.health - damage;
+  if (this.health >= 1) {
+    return "A Saxon has received " + damage + " points of damage";
+  } else {
+    return "A Saxon has died in combat";
+  }
+}
 
 // War
-function War() {}
+function War() {
+  this.vikingArmy = [];
+  this.saxonArmy = [];
+  this.addViking = function(viking){
+    this.vikingArmy.push(viking);
+  };
+  this.addSaxon = function(saxon){
+    this.saxonArmy.push(saxon);
+  };
+  this.vikingAttack = function(){
+    attackedSaxon = Math.floor(Math.random() * this.saxonArmy.length);
```

Comment:
this next lines and the ones in the next method are very similar, maybe we could encapsulate that in a function and reused in both methods :)